compute_tools/src/http/routes/configure.rs

```diff
@@ -85,3 +91,47 @@ pub(in crate::http) async fn configure(
 
     JsonResponse::success(StatusCode::OK, body)
 }
+
+// Accept spec in JSON format and validate the configuration
+// At the moment the validation is limited to the connstrings vector
+// for the safekeeper, and we only check that we have at least one entry
+// in there and no duplicate values
+pub(in crate::http) async fn check(
+    State(_compute): State<Arc<ComputeNode>>,
+    request: Json<ConfigurationRequest>,
+) -> Response {
+    let pspec = match ParsedSpec::try_from(request.spec.clone()) {
+        Ok(p) => p,
+        Err(e) => return JsonResponse::error(StatusCode::BAD_REQUEST, e),
+    };
+
+    match check_config(&pspec) {
+        Ok(b) => b,
+        Err(e) => return JsonResponse::error(StatusCode::BAD_REQUEST, e),
+    };
+
+    JsonResponse::success(StatusCode::OK, true)
+}
+
+// check an already parsed configuration
+fn check_config(pspec: &ParsedSpec) -> Result<bool, &'static str> {
```

Comment:
I think this should be a method on `ParsedSpec` struct, so we can call it in `fn try_from(spec: ComputeSpec)`, e.g. at the end of `try_from`. So that validation automatically applies to all call-sites, where `ParsedSpec` is produced

Reply:
Done that way now.